constraints/src/main/java/gov/nasa/jpl/aerie/constraints/time/Spans.java

```diff
@@ -7,65 +7,108 @@
 import java.util.Iterator;
 import java.util.List;
 import java.util.Objects;
+import java.util.Optional;
 import java.util.function.Function;
 import java.util.function.Predicate;
 import java.util.stream.Stream;
 import java.util.stream.StreamSupport;
 
 import gov.nasa.jpl.aerie.constraints.time.Interval.Inclusivity;
+import org.apache.commons.lang3.tuple.Pair;
 
 /**
  * A collection of intervals that can overlap.
  */
-public class Spans implements IntervalContainer<Spans>, Iterable<Interval> {
-  private final List<Interval> intervals;
+public class Spans implements IntervalContainer<Spans>, Iterable<Pair<Interval, Optional<Spans.Metadata>>> {
+  private final List<Pair<Interval, Optional<Metadata>>> intervals;
+
+  public record Metadata(long id){}
```

Comment:
I think this needs to support a list of ids. Otherwise we can't provide both activities in a violation of a mutex constraint.

Reply:
@adrienmaillard I guess this can be safely ignored for now, since we reduced the scope.